### ISSUE TITLE
[java] ImmutableField: Do not flag fields annotated with @Inject

### ISSUE THREAD

**Issue by maikelsteneker** [pmd/pmd#4011]
**Affects PMD Version:** 6.46.0

**Rule:** [ImmutableField](https://pmd.github.io/latest/pmd_rules_java_design.html#immutablefield)

**Description:** Fields that are annotated using the `javax.inject.Inject` annotation should not be final. However, a violation suggesting as much is still triggered.

**Code Sample demonstrating the issue:**

```
public abstract class AbstractDapiProcess implements Process {
   @Inject private AuthenticationService authService;
```

**Expected outcome:** No violation

PMD reports a violation at line 2 but that's wrong. That's a false positive.

**Running PMD through:** *CLI*
